### ISSUE TITLE
to train with my own data ?

### ISSUE THREAD

**Issue by DelphIONe** [biodlab/RODAN#7]
Hi,

I have basecalled my data with RODAN and I have compared the mapping result with guppy's basecalling but it's not better. So, I was wondering if is it possible to use model.py with a modified config file with my data preformatted by taiyaki (after prepare_mapped_reads step) ? I looked at your hdf5 structure but it seems different from the Taiyaki file.
Is it to possible to convert this file to launch a training by RODAN or not ?
If I understand correctly, the valid file (rna-valid.hdf5) would be my taiyaki result (from prepare_mapped_read step) and my training file (rna-train.hdf5) would be my reads? But how can I get this file from the fast5 of my reads? If you can help me, probably there is something I don't understand.
I'm testing your command ./model.py -c rna.config -n NAME -l, in theory the resulting model (what you call configuration (?)) should be the same as the one used when basecalling with RODAN. Is it correct ? Just to be sure because I don't understand the point of such a command if we don't learn from your own data.

Thanks for your time and sorry if my questions are basic questions.

**Comment by DelphIONe** [comment]
?

**Comment by biodlab** [comment]
The rna-train and rna-valid files are a special format for the dataloader which was generated from taiyaki prepared training data using the gendata.py script available in the repository. You can run it like this:

`$ gendata.py -i taiyaki.hdf5`

This will generate train.hdf5 and valid.hdf5 in the current working directory. I would reccomend at least 1M samples for training and 32k samples for validation.

**Comment by DelphIONe** [comment]
Thanks for your reply.
I have launched gendata.py as you mentioned
./gendata.py -i /home/xxx/outputTayaki.fast5 --outdir /home/xxx/rodan_gendata
but I obtain only the train.hdf5 file (without error). Any idea ?

**Comment by biodlab** [comment]
That's because there isn't enough samples in your original taiyaki data file to generate the pre-set amount of train and validation samples. The default is 1M training samples, and 100k validation samples.  I would recommend at least 1M training and 32k validation samples. You will need to reduce the amount of training and validation samples with the following options:

```
--maxchunks # (for training samples)
--maxvalchunks # (for validation samples)
```

You can also run:
```
$ h5ls -r train.hdf5
/                        Group
/events                  Dataset {2000000, 4096}
/labels                  Dataset {2000000, 204}
/labels_len              Dataset {2000000}
/status                  Dataset {2000000}

```
to see how many records total were created so you can figure out how many chunks are available from the taiyaki hdf5 file. Here we can see there are 2M samples in the train file.

**Comment by DelphIONe** [comment]
Thanks for this answer!
h5ls train.hdf5 
events                   Dataset {12426/Inf, 4096}
labels                   Dataset {12426/Inf, 99/4096}
labels_len               Dataset {12426/Inf}

Probably, I don't have enough reads... Can you explain the "Inf" value ? Or "99/4096"?
Thanks again,

**Comment by biodlab** [comment]
Unfortunately, 12k reads is not enough to produce the necessary amount of training data.
